testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/cli/CLIEmbedHostControllerTestCase.java

```diff
@@ -472,6 +480,294 @@ public void testPrivateHiddenOperations() throws IOException {
+        cli.sendLine(":read-children-names(child-type=host)");
+        final CLIOpResult result = cli.readAllAsOpResult();
+        final ModelNode response = result.getResponseNode();
+        assertTrue(response.has(OUTCOME));
+        assertEquals(SUCCESS, response.get(OUTCOME).asString());
+        assertTrue(response.has(RESULT));
+        // whatever the unpersisted hostname is, it shouldn't be the one we added.
+        assertNotEquals("foo33f", response.get(RESULT).asList().get(0).asString());
+        assertNotEquals(0, response.get(RESULT).asList().size());
+        assertTrue(cli.isConnected());
+    }
+
+    @Test
+    public void testEmptyDomainConfigConfigureWithReload() throws Exception {
+        final String line = "embed-host-controller --std-out=echo --domain-config=domain-empty-cli.xml --empty-domain-config --remove-existing-domain-config" + JBOSS_HOME;
+        cli.sendLine(line);
+        assertTrue(cli.isConnected());
+        cli.sendLine("/interface=test-interface:add(any-address)");
+        cli.sendLine("/socket-binding-group=test-sbg:add(default-interface=test-interface)");
+        cli.sendLine("/profile=test-profile:add()");
+        cli.sendLine("/server-group=test-server-group:add(profile=test-profile, socket-binding-group=test-sbg)");
+
+        cli.sendLine("ls /server-group=test-server-group");
+        cli.sendLine("reload --host=master --admin-only=true");
+        assertTrue(cli.isConnected());
+        cli.sendLine("ls /interface=test-interface");
+        cli.sendLine("ls /socket-binding-group=test-sbg");
+        cli.sendLine("ls /profile=test-profile");
+        cli.sendLine("ls /server-group=test-server-group");
+    }
+
+    @Test
+    public void testEmptyDomainConfigAndHostConfigConfigureWithReload() throws Exception {
+        final String line = "embed-host-controller --std-out=echo --domain-config=domain-empty-cli.xml --empty-domain-config --remove-existing-domain-config --host-config=host-empty-cli.xml --empty-host-config --remove-existing-host-config" + JBOSS_HOME;
+        cli.sendLine(line);
+        assertTrue(cli.isConnected());
+
+        // host-empty-cli.xml
+        assertTrue(cli.isConnected());
+        cli.sendLine("/host=master1:add()");
+        cli.sendLine("/host=master1/core-service=management/security-realm=test-realm:add()");
+        cli.sendLine("/host=master1/core-service=management/security-realm=test-realm/authentication=local:add(default-user=\"$local\", skip-group-loading=true)");
+        cli.sendLine("/host=master1/interface=test-mgmt-interface:add(inet-address=${jboss.bind.address.management:127.0.0.1})");
+        cli.sendLine("/host=master1/core-service=management/management-interface=http-interface:add(interface=test-mgmt-interface, security-realm=test-realm, http-upgrade-enabled=true, port=${jboss.management.http.port:9990})");
+
+        // reload after host added to configure the domain
+        cli.sendLine("reload --host=master1 --admin-only=true");
```

Comment:
Is this reload required? (Please ping me in HC with the answer, rather than here where I'll miss it.)